tools/system_libs.py

```diff
@@ -311,7 +311,8 @@ def build_objects(self, build_dir):
       else:
         cmd = [shared.EMXX]
 
-      cmd += cflags
+      if ext != '.s':
+        cmd += cflags
```

Comment:
What is the difference between `.S` and `.s` files? (I guess that difference explains the cflags handling difference?)

Reply:
`.S` files go through the pre-processor and so flags like `-D` and `-I` are valid. `-s` files are fed directly the assembler.

Reply:
Added a comment